### PR TITLE
EUI-4210-from-release-4.3.0

### PR DESCRIPTION
**Pull request by RiteshHMCTS**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/EUI-4210

### Change description ###
Automated test fails

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/shared/components/activity/activity.component.ts [hunk 1]
```diff
@@ -70,9 +70,11 @@ export class ActivityComponent implements OnInit, OnDestroy {
     return (this.activity.editors.length > 0 || this.activity.unknownEditors > 0)
   }
 
-  ngOnDestroy() {
-    this.subscription.complete();
-    this.subscription.unsubscribe();
+  ngOnDestroy(): void {
```
Comment (kuda1992) [inline src/shared/components/activity/activity.component.ts:73]: visibility modifier missing.

File: src/shared/components/case-editor/case-edit-form/case-edit-form.component.ts [hunk 2]
```diff
@@ -28,17 +28,23 @@ export class CaseEditFormComponent implements OnDestroy, AfterViewInit {
 
   constructor(private formValueService: FormValueService) {}
 
-  ngOnDestroy() {
-    this.pageChangeSubscription.unsubscribe();
-    this.formGroupChangeSubscription.unsubscribe();
+  ngOnDestroy(): void {
```
Comment (kuda1992) [inline src/shared/components/case-editor/case-edit-form/case-edit-form.component.ts:31]: Again visibility modifier missing

File: src/shared/components/case-history/case-history.component.ts [hunk 3]
```diff
@@ -63,8 +63,10 @@ export class CaseHistoryComponent implements OnInit, OnDestroy {
     });
   }
 
-  ngOnDestroy() {
-    this.caseSubscription.unsubscribe();
+  ngOnDestroy(): void {
```
Comment (kuda1992) [inline src/shared/components/case-history/case-history.component.ts:66]: Again visibility modifier missing.

**Review by leepavlouhmcts — DISMISSED** [review]

File: src/shared/services/activity/activity.service.ts [hunk 4]
```diff
@@ -47,23 +47,34 @@ export class ActivityService {
...
-    const url = this.activityUrl() + `/cases/${caseId.join(',')}/activity`;
-    return this.http
-      .get(url, options, false, ActivityService.handleHttpError)
-      .map(response => response);
+    try {
+      const options = this.getOptions();
+      const url = this.activityUrl() + `/cases/${caseId.join(',')}/activity`;
+      return this.http
+        .get(url, options, false, ActivityService.handleHttpError)
+        .map(response => response);
```
Comment (kuda1992) [inline src/shared/services/activity/activity.service.ts:55]: What is the use of the map? We are not transforming the response at all. Please remove!

**Review by johnbenjamin-hmcts — DISMISSED** [review]

File: src/shared/services/activity/activity.service.ts [hunk 5]
```diff
@@ -47,23 +47,34 @@ export class ActivityService {
...
+      return this.http
+        .post(url, body, options, false)
+        .map(response => response);
+    } catch (error) {
+      console.log('user may not be authenticated.' + error);
+    }
   }
 
   public verifyUserIsAuthorized(): void {
+    if (this.sessionStorageService.getItem('userDetails') === undefined) {
```
Comment (kuda1992) [inline src/shared/services/activity/activity.service.ts:75]: Why are  we checking explicitly for undefined. Sessionstorage return null if userDetails is not in local storage.  
```suggestion
    if (!this.sessionStorageService.getItem('userDetails')) {
```

**Review by kuda1992 — CHANGES_REQUESTED** [review]
Left a few comments

**Review by leepavlouhmcts — DISMISSED** [review]

**Review by johnbenjamin-hmcts — DISMISSED** [review]

File: src/shared/services/activity/activity.service.ts (hunk 5 above)
Comment (RiteshHMCTS) [reply src/shared/services/activity/activity.service.ts:75]: @kuda1992 I tried using that before I changes it to this. I also not sure why its behaving like this

File: src/shared/components/case-editor/case-edit-form/case-edit-form.component.ts (hunk 2 above)
Comment (RiteshHMCTS) [reply src/shared/components/case-editor/case-edit-form/case-edit-form.component.ts:31]: added

File: src/shared/components/activity/activity.component.ts (hunk 1 above)
Comment (RiteshHMCTS) [reply src/shared/components/activity/activity.component.ts:73]: added

File: src/shared/components/case-history/case-history.component.ts (hunk 3 above)
Comment (RiteshHMCTS) [reply src/shared/components/case-history/case-history.component.ts:66]: added

File: src/shared/services/activity/activity.service.ts (hunk 4 above)
Comment (RiteshHMCTS) [reply src/shared/services/activity/activity.service.ts:55]: for time being I can't remove it as it is already in production, but Uday has mentioned some changes yesterday so when we implement that change we can get rid of this.

File: src/shared/services/activity/activity.service.ts (hunk 5 above)
Comment (kuda1992) [reply src/shared/services/activity/activity.service.ts:75]: @RiteshHMCTS I don't get what you're trying to say. What's behaving like this?

Comment (kuda1992) [reply src/shared/services/activity/activity.service.ts:75]: Thanks for getting back @RiteshHMCTS I will approve

**Review by kuda1992 — APPROVED** [review]
All good

**Review by johnbenjamin-hmcts — APPROVED** [review]

**Pull request merged**